cyral/core/default_context_handler.go

```diff
@@ -0,0 +1,103 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/cyralinc/terraform-provider-cyral/cyral/client"
+	ot "github.com/cyralinc/terraform-provider-cyral/cyral/core/types/operationtype"
+	rt "github.com/cyralinc/terraform-provider-cyral/cyral/core/types/resourcetype"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// Implementation of a default context handler that can be used by all resources
+// which API follows these principles:
+//  1. The resource is backed by an ID coming from the API.
+//  2. The creation is a POST that returns a JSON with an `id` field, meaning
+//     it can be used with the `IDBasedResponse` struct.
+//  3. The endpoint to perform GET, PUT and DELETE calls are composed by the
+//     PUT endpoint plus the ID specification like the following:
+//     - POST:    https://<CP>/<apiVersion>/<featureName>
+//     - GET:     https://<CP>/<apiVersion>/<featureName>/<id>
+//     - PUT:     https://<CP>/<apiVersion>/<featureName>/<id>
+//     - DELETE:  https://<CP>/<apiVersion>/<featureName>/<id>
+type DefaultContextHandler struct {
+	ResourceName        string
+	ResourceType        rt.ResourceType
+	SchemaReaderFactory SchemaReaderFactoryFunc
+	SchemaWriterFactory SchemaWriterFactoryFunc
+	BaseURLFactory      URLFactoryFunc
+}
+
+func defaultSchemaWriterFactory(d *schema.ResourceData) SchemaWriter {
+	return &IDBasedResponse{}
+}
+
+func defaultOperationHandler(
+	resourceName string,
+	resourceType rt.ResourceType,
+	operationType ot.OperationType,
+	baseURLFactory URLFactoryFunc,
+	httpMethod string,
+	schemaReaderFactory SchemaReaderFactoryFunc,
+	schemaWriterFactory SchemaWriterFactoryFunc) ResourceOperationConfig {
+	// GET = https://<CP>/<apiVersion>/<feature>
+	// POST, PUT and DELETE = https://<CP>/<apiVersion>/<feature>/<repoId>
+	endpoint := func(d *schema.ResourceData, c *client.Client) string {
+		url := ""
+		if d.Id() == "" {
+			url = baseURLFactory(d, c)
+		} else {
+			url = fmt.Sprintf("%s/%s", baseURLFactory(d, c), d.Id())
+		}
```

Comment:
Suggestion:
```suggestion
		url := baseURLFactory(d, c)
		if d.Id() != "" {
			url = fmt.Sprintf("%s/%s", url, d.Id())
		} 
```

Reply:
Fixed in [34c2a8d](https://github.com/cyralinc/terraform-provider-cyral/pull/474/commits/34c2a8d948c132bad8b2d953865472c57ce671f1).